File: Maria Mercedes Atim/JAVASCRIPT/Clase 10/Producto.js
class Producto {
    static contadorProductos = 0;

    constructor(nombre, precio) {
        this._idProducto = ++Producto.contadorProductos;
        this._nombre = nombre;
        this._precio = precio;
    }

    // Método para obtener el ID del producto
    getIdProducto() {
        return this._idProducto;
    }

    // Método para obtener el nombre del producto
    getNombre() {
        return this._nombre;
    }

    // Método para establecer el nombre del producto
    setNombre(nombre) {
        this._nombre = nombre;
    }

    // Método para obtener el precio del producto
    getPrecio() {
        return this._precio;
    }

    // Método para establecer el precio del producto
    setPrecio(precio) {
        if (precio >= 0) {
            this._precio = precio;
        } else {
            console.error("El precio no puede ser negativo.");
        }
    }

    // Método toString
    toString() {
        return `Producto ID: ${this._idProducto}, Nombre: ${this._nombre}, Precio: ${this._precio}`;
    }
}

let producto1 = new Producto("Pantalon", 200);
let producto2 = new Producto("Camisa", 150);
console.log(producto1.toString());
console.log(producto2.toString());

File: Java/Clase12/Clase12/src/ar/com/codesystem/ventas/Producto.java
package ar.com.codesystem.ventas;

public class Producto {
    // Atributos
    private int idProducto;
    private String nombre;
    private double precio;
    private static int contadorProductos;

    //Constructor vacio
    private Producto() {
        this.idProducto = ++Producto.contadorProductos;
    }

    public Producto(String nombre, double precio) {
        this();
        this.nombre = nombre;
        this.precio = precio;
    }

    public int getIdProducto() {
        return idProducto;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public double getPrecio() {
        return precio;
    }

    public void setPrecio(double precio) {
        this.precio = precio;
    }

    @Override // Sobreescribimos el toString() para mostrar la cadena de producto
    public String toString() {
        return "Producto { " + "idProducto: " + idProducto + ", nombre: " + nombre + ", precio: " + precio + " }";
    }
}